packages/driver/src/cypress/log.ts

```diff
@@ -524,36 +537,41 @@ function create (Cypress, cy, state, config) {
     }
   }
 
-  const triggerLog = function (log) {
+  triggerLog (log) {
     log._hasInitiallyLogged = true
 
-    return trigger(log, 'command:log:added')
+    return this.trigger(log, 'command:log:added')
   }
 
-  const addToLogs = function (log) {
+  addToLogs (log) {
     const id = log.get('id')
 
-    logs[id] = true
+    this.logs[id] = true
   }
 
-  const logFn = function (options: any = {}) {
-    if (!_.isObject(options)) {
-      $errUtils.throwErrByPath('log.invalid_argument', { args: { arg: options } })
+  // only fire the log:state:changed event
+  // as fast as every 4ms
+  fireChangeEvent (log) {
+    const triggerStateChanged = () => {
+      return this.trigger(log, 'command:log:changed')
     }
 
-    const log = Log(cy, state, config, options)
+    const debounceFn = _.debounce(triggerStateChanged, 4)
 
-    // add event emitter interface
-    $Events.extend(log)
+    return debounceFn()
+  }
+}
 
-    const triggerStateChanged = () => {
-      return trigger(log, 'command:log:changed')
+export function create (Cypress, cy, state, config) {
+  counter = 0
+  const logManager = new LogManager()
+
+  const logFn = function (options: any = {}) {
+    if (!_.isObject(options)) {
+      $errUtils.throwErrByPath('log.invalid_argument', { args: { arg: options } })
     }
 
-    // only fire the log:state:changed event
-    // as fast as every 4ms
-    // @ts-ignore
-    log.fireChangeEvent = _.debounce(triggerStateChanged, 4)
+    const log = new Log(cy, state, config, logManager, options)
```

Comment:
Are we able to pass `logManager.fireChangeEvent` instead of the entire class instance? Since the logManger stores all of the incoming logs, this feels heavy to each log instance.

Reply:
Done.